cmd/frontend/graphqlbackend/oobmigrations.go

```diff
@@ -43,6 +48,33 @@ func (r *schemaResolver) OutOfBandMigrations(ctx context.Context) ([]*outOfBandM
 		return nil, err
 	}
 
+	if args.FilterDeprecatedBeforeFirstVersion != nil && *args.FilterDeprecatedBeforeFirstVersion {
+		rawFirstVersion, ok, err := upgradestore.New(r.db).GetFirstServiceVersion(ctx)
+		if err != nil {
+			return nil, err
+		}
+		if ok {
+			firstVersion, ok := oobmigration.NewVersionFromString(rawFirstVersion)
+			if !ok {
+				return nil, errors.Newf("invalid first version %q", rawFirstVersion)
+			}
+			filtered := migrations[:0]
+			for _, migration := range migrations {
+				// If the migration has no deprecated value, append it to the list.
+				if migration.Deprecated == nil {
+					filtered = append(filtered, migration)
+					continue
+				}
+
+				// If the migration is deprecated before the first version of sourcegraph, don't append it to the list.
+				if oobmigration.CompareVersions(firstVersion, *migration.Deprecated) != oobmigration.VersionOrderAfter {
+					filtered = append(filtered, migration)
```

Comment:
I understand it would be redundant in the current form of this function/loop, but adding this `continue` now would give ourselves an easier life in the future. Because at a later point, we would have to think really hard that "this did not have continue was because we shouldn't, or it did not matter?"

```suggestion
					filtered = append(filtered, migration)
					continue
```